src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,356 @@
+
+    /**
+     * Get the plugin installation method.
+     *
+     * @param string Plugin name
+     * @return string Plugin installation method
+     */
+    protected function getInstallMethod($plugin)
+    {
+        $slash = $this->getSlash();
+        $plugin_dir = $this->getPluginDir($plugin);
+        $git_dir = $plugin_dir . $slash . '.git';
+        if (is_dir("$git_dir") && $this->commandExists('git')) {
+            return 'git';
+        }
+        $composer_json = $plugin_dir . $slash . 'composer.json';
+        if (file_exists($composer_json)) {
+            return $this->commandExists('composer') ? 'composer' : 'archive';
+        }
+        return 'unknown';
+    }
+
+    /**
+     * Update a specific plugin.
+     *
+     * @param string $plugin Plugin name
+     */
+    protected function updatePlugin($plugin)
+    {
+        $plugins_dir = $this->getPluginDir();
+        $plugin_dir = $plugins_dir . $plugin;
+        if (!is_dir("$plugin_dir")) {
+            $message = "{$plugin} is not installed.";
+            throw new TerminusNotFoundException($message);
+        }
+        $messages = array();
+        $message = "Updating {$plugin}...";
+        $this->log()->notice($message);
+        $method = $this->getInstallMethod($plugin);
+        switch ($method) {
+            case 'git':
+                // Compare installed version to the latest release.
+                $installed_version = $this->getInstalledVersion($plugin_dir);
+                $latest_version = $this->getLatestVersion($plugin_dir);
+                if ($installed_version != 'unknown' && $latest_version != 'unknown') {
+                    if ($installed_version < $latest_version) {
+                        exec("cd \"$plugin_dir\" && git checkout {$latest_version}", $messages);
+                    } else {
+                        $messages[] = "Already up-to-date.";
+                    }
+                } else {
+                    $messages[] = "Unable to update. Semver compliance issue with tagged release.";
+                }
+                break;
+
+            case 'composer':
+                // Determine the project name.
+                $composer_info = $this->getComposerInfo($plugin);
+                $project = $composer_info['name'];
+                $packagist_url = "https://packagist.org/packages/{$project}";
+                if ($this->isValidUrl($packagist_url)) {
+                    // Backup the plugin directory, just in case.
+                    $datetime = date('YmdHi', time());
+                    $backup_directory = "~/.composer/backups/{$project}/{$datetime}";
+                    exec("mkdir -p {$backup_directory} && tar czvf {$backup_directory}/backup.tar.gz \"{$plugin_dir}\"", $backup_messages);
+                    // Create a new project via Composer.
+                    exec("rm -rf \"{$plugin_dir}\" && composer create-project --prefer-source --keep-vcs -n -d {$plugins_dir} {$project}:~1", $install_messages);
```

Comment:
Good call.  Missed that.  I'll try and push up another commit soon that fixes this and a few other issues.